dwave/optimization/model.pxd

```diff
@@ -19,7 +19,8 @@ from libcpp.memory cimport shared_ptr
 from libcpp.vector cimport vector
 
 from dwave.optimization.libcpp.array cimport Array as cppArray
```

Comment:
`cppArray` unused?

Reply:
Moved to the `.pyx`. Annoyingly I cannot just use `cppArrayNode` everywhere because Cython gets confused about inheritance and overloads.